### DISCUSSION TITLE
Way to silence "warning Query takes too long"?

### DISCUSSION THREAD

**Discussion by samajammin** [gatsbyjs/gatsby#30767] [Ideas / Feature Requests]
Hey folks - I believe we're having the same GraphQL issue as here:
https://github.com/gatsbyjs/gatsby/issues/30256

Perhaps this is a bandaid solution but I'm exploring options / requesting a feature to be able to silence these warnings:
https://github.com/ethereum/ethereum-org-website/issues/2810

Specifically, we'd like to silence the logging of page context:
https://github.com/gatsbyjs/gatsby/blob/54d4721462b9303fed723fdcb15ac5d72e103778/packages/gatsby/src/query/query-runner.ts#L41

Reason being, we use the [`gatsby-plugin-intl`](https://github.com/wiziple/gatsby-plugin-intl) community plugin, which stores all our content strings in message data in the context object. Whenever this warning logs the context, this results in >1K lines of output, which makes our build logs nearly impossible to sift through (& often breaks the build logs UI in Netlify).

Would it be possible to add some ENV variable or feature flag to `reportLongRunningQueryJob` so that we could silence this logs?

Thanks in advance!

**Comment by redabacha** [comment]
i'm having this issue also. i've made a pr https://github.com/gatsbyjs/gatsby/pull/34279 to add an option for this. in the meantime i'm using this dirty fix in my `gatsby-config.js`:
```js
const reporter = require('gatsby-cli/lib/reporter');

// monkey patch reporter.warn to avoid spam when reporting long running queries
const originalReporterWarn = reporter.warn;
reporter.warn = text => {
  if (
    text.includes('Query takes too long') ||
    text.includes('This query took more than 15s to run')
  ) {
    return originalReporterWarn(text.replace(/Context:.*/s, ''));
  }

  return originalReporterWarn(text);
};
```

**Comment by a-botermans** [comment]
As per @KyleAMathews request on discord I'm adding info to this thread. For me @redabacha solution didn't work. I'm on Gatsby 3.14.6 (I know upgrading would probably help build time but it's not trivial). We were timing out on Netlify with build of 1h for now we solved the problem by redirecting all output:

`gatsby build > /dev/null`

This is obviously not a long term solution as it means we have zero view on the logs. But it sped up the build process a lot now the build takes ~10 minutes. Printing the entire context can lead to huge amounts of logs depending on the site.

**Reply by a-botermans** [reply]
Upgrading to 4.24.8  made @redabacha solution work :)
